lib/mzgl/gl/api/OpenGLAPI.h

```diff
@@ -0,0 +1,337 @@
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+STRINGIFY(
+
+			in lowp vec4 colorV;
+			in lowp vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main(void) {
+				fragColor = texture(myTextureSampler, texCoordV) * colorV;
+			}
+
+		));
+
+		g.fontShader = Shader::create(g);
+		g.fontShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+
+			in vec4 Position;
+			in vec2 TexCoord;
+
+			uniform lowp vec4 color;
+
+			out vec2 texCoordV;
+
+			void main() {
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+STRINGIFY(
+
+			uniform lowp vec4 color;
+			in vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main() {
+				fragColor = color;
+				fragColor.a *= texture(myTextureSampler, texCoordV).a;
+			}
+
+		));
+		g.texShader = Shader::create(g);
+		g.texShader->loadFromString(STRINGIFY(
+
+			uniform mat4 mvp;
+
+			in vec4 Position;
+			in lowp vec2 TexCoord;
+			uniform lowp vec4 color;
+
+			out lowp vec4 colorV;
+			out lowp vec2 texCoordV;
+			void main(void) {
+				colorV		= color;
+				texCoordV	= TexCoord;
+				gl_Position = mvp * Position;
+			}
+
+		),
+	STRINGIFY(
+
+			in lowp vec4 colorV;
+			in lowp vec2 texCoordV;
+			out vec4 fragColor;
+			uniform sampler2D myTextureSampler;
+
+			void main(void) {
+				fragColor = texture(myTextureSampler, texCoordV) * colorV;
+			}
+
+		));
+		// clang-format on
+
+		g.nothingShader->isDefaultShader	  = true;
+		g.colorShader->isDefaultShader		  = true;
+		g.colorTextureShader->isDefaultShader = true;
+		g.texShader->isDefaultShader		  = true;
+		g.fontShader->isDefaultShader		  = true;
+	}
+
+	void setLineWidth(float f) override { glLineWidth(f); }
+	void setBlending(bool shouldBlend) override {
+		if (shouldBlend) {
+			glEnable(GL_BLEND);
+		} else {
+			glDisable(GL_BLEND);
+		}
+	}
+
+	void drawVerts(const std::vector<glm::vec2> &verts, Vbo::PrimitiveType type) override {
+		g.nothingShader->begin();
+
+		g.currShader->uniform("mvp", g.getMVP());
+		if (g.currShader->needsColorUniform) {
+			g.currShader->uniform("color", g.getColor());
+		}
+
+#ifndef MZGL_GL2
+		if (immediateVertexArray == 0) return;
+		glBindVertexArray(immediateVertexArray);
+#endif
+
+		glEnableVertexAttribArray(g.currShader->positionAttribute);
+		glBindBuffer(GL_ARRAY_BUFFER, immediateVertexBuffer);
+		glBufferData(GL_ARRAY_BUFFER, verts.size() * 2 * sizeof(float), verts.data(), GL_DYNAMIC_DRAW);
+		glVertexAttribPointer(g.currShader->positionAttribute, 2, GL_FLOAT, GL_FALSE, 0, NULL);
+
+		auto glType = primitiveTypeToGLMode(type);
+
+		glDrawArrays(glType, 0, (GLsizei) verts.size());
+
+		glDisableVertexAttribArray(g.currShader->positionAttribute);
+
+#ifndef MZGL_GL2
+		glBindVertexArray(0);
+#endif
+	}
+
+	void drawVerts(const std::vector<glm::vec2> &verts,
+				   const std::vector<glm::vec4> &cols,
+				   Vbo::PrimitiveType type) override {
+		g.colorShader->begin();
```

Comment:
Im not a big fan of single letter variable names. I assume this is the graphics class here?

Reply:
I try to not have too many but Graphics is just used so much (as well as R) that I let those ones slide.